python_modules/dagster/dagster/core/execution.py

```diff
@@ -583,33 +589,89 @@ def construct_pipeline_execution_context(
     )
 
 
-def _create_loggers(run_config, execution_context):
+def _create_loggers(environment_config, run_config, pipeline_def):
+    check.inst_param(environment_config, 'environment_config', EnvironmentConfig)
     check.inst_param(run_config, 'run_config', RunConfig)
-    check.inst_param(execution_context, 'execution_context', ExecutionContext)
+    check.inst_param(pipeline_def, 'pipeline_def', PipelineDefinition)
+
+    loggers = []
+    for logger_key, logger_def in pipeline_def.loggers.items() or default_loggers().items():
```

Comment:
does this make it so that it is impossible to explicitly configure a pipeline to not have loggers?

as written it kind of looks like it will fatal (it's a bit odd to call .items() on an empty dict). seems like code could be slightly more obvious and look more bullet proof.

Reply:
```
>>> {}.items()
dict_items([])
```
...but I take your point and we can make this more obvious

Reply:
And yes, this is to make it impossible to not have loggers... the idea is that `default_loggers()` is all the loggers than you can configure on a pipeline that doesn't explicitly declare any, and `default_system_loggers()` is all the loggers and associated config that we use when no logger config is provided at all